bin/ch/Helpers.cpp

```diff
@@ -7,30 +7,41 @@
 #include <sys/stat.h>
 
 //TODO: x-plat definitions
-#ifdef _WIN32
 #define MAX_URI_LENGTH 512
+#ifdef _WIN32
+#define TTD_MAX_FILE_LENGTH MAX_PATH
 #define TTD_HOST_PATH_SEP "\\"
+#else
+#define TTD_MAX_FILE_LENGTH MAX_URI_LENGTH
+#define TTD_HOST_PATH_SEP "/"
+#endif
 
 void TTDHostBuildCurrentExeDirectory(char* path, size_t* pathLength, size_t bufferLength)
 {
-    wchar exePath[MAX_PATH];
-    GetModuleFileName(NULL, exePath, MAX_PATH);
+    char exePath[TTD_MAX_FILE_LENGTH];
+    PlatformAgnostic::SystemInfo::GetBinaryLocation(exePath, TTD_MAX_FILE_LENGTH);
 
-    size_t i = wcslen(exePath) - 1;
-    while(exePath[i] != _u('\\'))
+    size_t i = strlen(exePath) - 1;
+    while (exePath[i] != TTD_HOST_PATH_SEP[0] && i != 0)
     {
         --i;
     }
-
-    if(i * 3 > bufferLength)
+    if (i == 0)
     {
-        printf("Don't overflow path buffer during conversion");
+        printf("%s", exePath);  // If exePath doesn't contain path separator, it's an error message
```

Comment:
I mean this line doesn't print a path. `exePath` itself is an error message.
The `GetBinaryLocation` function sets `exePath` to either an actual path or an error message. (such as [L52 of SystemInfo.h](https://github.com/xiaoyinl/ChakraCore/blob/eed45f6ae78e004502614afc34edefd5367a325a/lib/Common/PlatformAgnostic/SystemInfo.h#L52))
Here the `while` loop finds the index of the path separator ("\\" or "/") in `exePath`. If `i == 0`, the path separator doesn't exist, so `exePath` is not a path, but an error message.